DSESwift/DataSyncer.swift

```diff
@@ -8,37 +8,398 @@
 
 import Foundation
 import RealmSwift
+import PromiseKit
+import CoreLocation
+import SwiftyJSON
 
-class DataSyncer: NSObject {
 
-    func login() {
-    downloadSensorProfile ()
+
+/**
+ * DataSyncer handles the synchronization between the local storage and CommonSense.
+ * The syncing process is handled automatically and periodically, thus the external
+ * user does not need to be aware of the data syncing process at all.
+ *
+ */
+public class DataSyncer : NSObject {
```

Comment:
does not have to be a public class right? we do not expose it to the end user. Public goes outside of the DSE scope.